encoding/unmarshaler.go

```diff
@@ -43,9 +46,21 @@ func (u *Unmarshaler) Unmarshal(dest interface{}) error {
 			return err
 		}
 
+		if u.isBodyless(next) {
+			continue
+		}
+
 		val := reflect.ValueOf(next).Elem()
 		field.Set(val.Convert(field.Type()))
 	}
 
 	return nil
 }
+
+// isBodyless returns a bool representing whether or not the given amf0.Type is
+// bodyless or not.
+func (u *Unmarshaler) isBodyless(t amf0.AmfType) bool {
+	_, isBodyless := t.(amf0.Bodyless)
+
+	return isBodyless
```

Comment:
Sure, I'll make that change. The idea behind the `Bodyless` interface is that it is a "tag type", meaning that the actual value that the method returns is meaningless: the method simply exists to implement the type. In my opinion, the "tag type" is the second weakest abstraction, just above `interface{}` (but this is another discussion entirely).

Anyway, it probably makes sense that all implementers of the `Bodyless` return `true` for `IsBodyless`, so I'll make the change regardless. :smile: